crates/blockchain/payload.rs

```diff
@@ -96,7 +101,19 @@ pub fn build_payload(args: &BuildPayloadArgs, storage: &Store) -> Result<Block,
             ommers: Vec::new(),
             withdrawals: Some(args.withdrawals.clone()),
         },
-    })
+    };
+    // Apply withdrawals & call beacon root contract, and obtain the new state root
+    let spec_id = spec_id(storage, args.timestamp)?;
+    let mut evm_state = evm_state(storage.clone(), parent_block.number);
+    if args.beacon_root.is_some() && spec_id == SpecId::CANCUN {
+        beacon_root_contract_call(&mut evm_state, &payload.header, spec_id)?;
```

Comment:
I feel like this should be done inside the evm crate. `spec_id` is specific to revm and should not be leaked outside ideally.